web3/transaction.py

```diff
@@ -0,0 +1,66 @@
+
+
+class Transaction(object):
```

Comment:
A few things.

1. Lets move this object to `web3.utils.datastructures`.
2. Rename to `AttributeDict` and remove any language that is transaction specific.
3. Remove the immutability functionality (will explain elsewhere)
3. Inherit the class from `collections.Mapping`